provisioning/baremetal_config.go

```diff
@@ -75,56 +78,129 @@ func getProvisioningNetworkMode(prov *metal3iov1alpha1.Provisioning) metal3iov1a
 }
 
 func validateManagedConfig(prov *metal3iov1alpha1.Provisioning) error {
-	for _, toTest := range []struct {
-		Name  string
-		Value string
-	}{
-
-		{Name: "ProvisioningInterface", Value: prov.Spec.ProvisioningInterface},
-		{Name: "ProvisioningIP", Value: prov.Spec.ProvisioningIP},
-		{Name: "ProvisioningNetworkCIDR", Value: prov.Spec.ProvisioningNetworkCIDR},
-		{Name: "ProvisioningDHCPRange", Value: prov.Spec.ProvisioningDHCPRange},
-		{Name: "ProvisioningOSDownloadURL", Value: prov.Spec.ProvisioningOSDownloadURL},
-	} {
-		if toTest.Value == "" {
-			return fmt.Errorf("%s is required but is empty", toTest.Name)
-		}
+	provisionConfig := map[string]string{
+		"ProvisioningInterface":     prov.Spec.ProvisioningInterface,
+		"ProvisioningIP":            prov.Spec.ProvisioningIP,
+		"ProvisioningNetworkCIDR":   prov.Spec.ProvisioningNetworkCIDR,
+		"ProvisioningDHCPRange":     prov.Spec.ProvisioningDHCPRange,
+		"ProvisioningOSDownloadURL": prov.Spec.ProvisioningOSDownloadURL,
 	}
-	return nil
+
+	return validateProvisioningConfig(provisionConfig)
 }
 
 func validateUnmanagedConfig(prov *metal3iov1alpha1.Provisioning) error {
-	for _, toTest := range []struct {
-		Name  string
-		Value string
-	}{
-
-		{Name: "ProvisioningInterface", Value: prov.Spec.ProvisioningInterface},
-		{Name: "ProvisioningIP", Value: prov.Spec.ProvisioningIP},
-		{Name: "ProvisioningNetworkCIDR", Value: prov.Spec.ProvisioningNetworkCIDR},
-		{Name: "ProvisioningOSDownloadURL", Value: prov.Spec.ProvisioningOSDownloadURL},
-	} {
-		if toTest.Value == "" {
-			return fmt.Errorf("%s is required but is empty", toTest.Name)
+	provisionConfig := map[string]string{
+		"ProvisioningInterface":     prov.Spec.ProvisioningInterface,
+		"ProvisioningIP":            prov.Spec.ProvisioningIP,
+		"ProvisioningNetworkCIDR":   prov.Spec.ProvisioningNetworkCIDR,
+		"ProvisioningOSDownloadURL": prov.Spec.ProvisioningOSDownloadURL,
+	}
+
+	return validateProvisioningConfig(provisionConfig)
+}
+
+func validateDisabledConfig(prov *metal3iov1alpha1.Provisioning) error {
+	provisionConfig := map[string]string{
+		"ProvisioningIP":            prov.Spec.ProvisioningIP,
+		"ProvisioningNetworkCIDR":   prov.Spec.ProvisioningNetworkCIDR,
+		"ProvisioningOSDownloadURL": prov.Spec.ProvisioningOSDownloadURL,
+	}
+
+	return validateProvisioningConfig(provisionConfig)
+}
+
+func validateProvisioningOSDownloadURL(uri string) error {
+	parsedURL, err := url.ParseRequestURI(uri)
+	if err != nil {
+		return fmt.Errorf("the URI provided: %s is invalid", uri)
+	}
+	if parsedURL.Scheme == "http" || parsedURL.Scheme == "https" {
+		var sha256Checksum string
+		if sha256Checksums, ok := parsedURL.Query()["sha256"]; ok {
```

Comment:
Is sha256 the only checksum type we support?

Reply:
This is what the installer expects and requests.  I do suppose we could support other hash types here since the user could request them.. thoughts?

Reply:
OK, if that's all we support for now then this is fine.